test/activerecord_spanner_mock_server/read_only_transaction_test.rb

```diff
@@ -111,9 +111,67 @@ def test_multiple_snapshots
       assert_empty rollback_requests
     end
 
-    def register_singer_find_by_id_result
-      sql = "SELECT `singers`.* FROM `singers` WHERE `singers`.`id` = @p1 LIMIT @p2"
-      @mock.put_statement_result sql, MockServerTests::create_random_singers_result(1)
+    def test_single_max_staleness
+      sql = register_singer_find_by_id_result true
+
+      Singer.optimizer_hints("max_staleness: 5.45").find_by(id: 1)
+
+      execute_sql_request = @mock.requests.find { |req| req.is_a?(Google::Cloud::Spanner::V1::ExecuteSqlRequest) && req.sql == sql }
+      single_use = execute_sql_request.transaction.single_use
+      assert single_use
+      assert single_use.read_only.return_read_timestamp
+      assert single_use.read_only.max_staleness
+      assert_equal 5, single_use.read_only.max_staleness.seconds
+      assert_equal 450000000, single_use.read_only.max_staleness.nanos
+    end
+
+    def test_single_exact_staleness
+      sql = register_singer_find_by_id_result true
+
+      Singer.optimizer_hints("exact_staleness:100").find_by(id: 1)
```

Comment:
the space before 100 would not have any difference/impact, right?

Reply:
Correct. I verified that by sometimes including a space and sometimes leaving it out in the test cases as well.